File: frontend/src/api/bills.ts
import client from './client'
import type { Bill, PaycheckConfig } from '../types'

export const getBills = () =>
  client.get<Bill[]>('/bills').then(r => r.data)

export const createBill = (data: { name: string; amount_cents: number; due_day: number }) =>
  client.post<Bill>('/bills', data).then(r => r.data)

export const deleteBill = (id: number) =>
  client.delete(`/bills/${id}`)

export const getPaycheckConfig = () =>
  client.get<PaycheckConfig | null>('/bills/paycheck').then(r => r.data)

export const savePaycheckConfig = (data: { reference_date: string; frequency_days: number; amount_cents: number }) =>
  client.post<PaycheckConfig>('/bills/paycheck', data).then(r => r.data)

File: backend/models/bills.py
from datetime import datetime, date
from sqlalchemy import Integer, String, DateTime, Date
from sqlalchemy.orm import Mapped, mapped_column
from ..database import Base


class Bill(Base):
    __tablename__ = "bills"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(Integer, nullable=False, index=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    amount_cents: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    due_day: Mapped[int] = mapped_column(Integer, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)


class PaycheckConfig(Base):
    __tablename__ = "paycheck_configs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(Integer, nullable=False, unique=True)
    reference_date: Mapped[date] = mapped_column(Date, nullable=False)
    frequency_days: Mapped[int] = mapped_column(Integer, nullable=False, default=14)
    amount_cents: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
